### ISSUE TITLE
PiHole restarts every couple minutes

### ISSUE THREAD

**Issue by rynomad** [samizdapp/skunkworks#1]
We have two pihole containers to implement split-horizon DNS, but this bug shows up even with only one. 

PiHole container does a shutdown and restart cycle every several minutes. This spams the logs and causes spotty performance. It works fine while it's up.

**Comment by HOGGL** [comment]
The is definitely a balenaOS issue as it does not occur with docker-compose. Maybe the balena supervisor is killing it because it doesn't enter `balena-idle`?

**Comment by HOGGL** [comment]
Ok so I ran `journalctl -f` and waited for the container to crash. And after a couple of minutes this error would be shown:

```
balenad[1089]: time="2022-04-11T23:37:53.176002346Z" level=info msg="Unhealthy container 85119926f78b[...]: restarting..."
```

So I investigated Docker 'heathchecks' and added `HEALTHCHECK NONE` to the end of `services/pihole/Dockerfile`. Now the container stays up!

---

When rebuilding the overwritten healthcheck command is shown:

 ```
CMD-SHELL dig +norecurse +retry=0 @127.0.0.1 pi.hole || exit 1
```

This is the same issue as #5. Inside a container, resolving with localhost times out. I just checked the available dns services with `ss -lu '( sport = :53 )'` and there is no localhost one. Outside of the container, there is a resolver available at localhost:53. This is demonstrated by running something like `nslookup $HOSTNAME 127.0.0.1`.

I think I read somewhere that balenaOS doesn't make available localhost:53 so that containers can have their own dns service.
